app/src/main/java/app/revanced/reddit/patches/SanitizeUrlQueryPatch.java

```diff
@@ -0,0 +1,9 @@
+package app.revanced.reddit.patches;
+
+public class UrlSanitizer {
+    public static String removeTrackingParameters(String url) {
+        Pattern pattern = Pattern.compile(".?utm_source=.+")
+        Matcher matcher = pattern.matcher(url)
+        return matcher.replaceAll("")
+    }
```

Comment:
Please test the changes, also instead of a regex, parse the URL and remove the query from the URL:
```suggestion
public final class SanitizeUrlQueryPatch {
	private static final Pattern SANITIZE_PATTERN = Pattern.compile(".?utm_source=.+");
    public static String removeTrackingParameters(final String url) {
        return SANITIZE_PATTERN.matcher(url).replaceAll("");
    }
```